terraform/staging/main.tf

```diff
@@ -0,0 +1,112 @@
+terraform {
+  required_version = "~> 1.3.6"
+
+  backend "s3" {
+    bucket         = "spack-terraform-state-staging"
+    key            = "terraform.tfstate"
+    region         = "us-west-1"
+    dynamodb_table = "spack-terraform-state-locks-staging"
+    encrypt        = true
+  }
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+    helm = {
+      source  = "hashicorp/helm"
+      version = "~> 2.5"
+    }
+    kubectl = {
+      source  = "gavinbunney/kubectl"
+      version = "~> 1.14"
+    }
+    flux = {
+      source  = "fluxcd/flux"
+      version = "~> 0.22.2"
+    }
+    github = {
+      source  = "integrations/github"
+      version = "~> 5.13.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = "us-west-1"
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = module.staging_cluster.cluster_endpoint
+    cluster_ca_certificate = module.staging_cluster.cluster_ca_certificate
+    exec {
+      api_version = "client.authentication.k8s.io/v1beta1"
+      args        = ["eks", "get-token", "--cluster-name", module.staging_cluster.cluster_name, "--region", "us-west-1"]
+      command     = "aws"
+    }
+  }
+}
+
+provider "kubectl" {
+  host                   = module.staging_cluster.cluster_endpoint
+  cluster_ca_certificate = module.staging_cluster.cluster_ca_certificate
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    args        = ["eks", "get-token", "--cluster-name", module.staging_cluster.cluster_name, "--region", "us-west-1"]
+    command     = "aws"
+  }
+}
+
+provider "kubernetes" {
+  host                   = module.staging_cluster.cluster_endpoint
+  cluster_ca_certificate = module.staging_cluster.cluster_ca_certificate
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    args        = ["eks", "get-token", "--cluster-name", module.staging_cluster.cluster_name, "--region", "us-west-1"]
+    command     = "aws"
+  }
+}
+
+
+provider "github" {
+  owner = "mvandenburgh" # TODO: change this
```

Comment:
Should this be changed as a part of this PR?

Reply:
Flux is still pointed at https://github.com/mvandenburgh/spack-infrastructure/tree/flux2, since we need a github personal access token from spackbot from Todd in order to configure it with `spack/spack-infrastructure`. I didn't want to block this PR on that so I just left it as-is.